### PR TITLE
[ntuple] fix importing leaf count arrays with broken title

### PR DESCRIPTION
**Pull request by jblomer**

### CODE REVIEW

File: tree/ntupleutil/v7/src/RNTupleImporter.cxx [hunk 1]
```diff
@@ -201,7 +201,10 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::PrepareSc
          }
 
          Int_t countval = 0;
-         auto *countleaf = l->GetLeafCounter(countval);
+         // We don't use GetLeafCounter() to retrieve the leaf count because it relies on the correct format of
+         // the leaf title. There are files in the public where the title is broken (empty).
+         l->GetLeafCounter(countval);
```
[the remark this answers is not in the document]
Comment (jblomer) [reply tree/ntupleutil/v7/src/RNTupleImporter.cxx]: Thank you, good catches!  I had to leave `GetLeafCounter` to determine if the leaf is a C string (see comments).

File: tree/ntupleutil/v7/src/RNTupleImporter.cxx [hunk 2]
```diff
@@ -173,6 +173,8 @@ ROOT::Experimental::RResult<void> ROOT::Experimental::RNTupleImporter::PrepareSc
          return R__FAIL("unsupported: count leaf arrays in leaf list, branch " + std::string(b->GetName()));
 
       // Only plain leafs with type identifies 'C' are C strings. Otherwise, they are char arrays.
+      // We use GetLeafCounter instead of GetLeafCount and GetLenStatic because the latter don't distinguish between
+      // char arrays and C strings.
```
Comment (pcanal) [inline tree/ntupleutil/v7/src/RNTupleImporter.cxx:177]: Lightly scanning the code of `GetLeafCounter` I don't exactly see where it 'distinguishes` between `char arrays` and `C strings`.  Can you expand the explanation a bit more?

Comment (jblomer) [reply tree/ntupleutil/v7/src/RNTupleImporter.cxx:177]: I think it's due to the different titles. C string read `myString/C` and char arrays `c[4]/C`.

**Review by jalopezg-git — APPROVED** [review]
Thanks!

**Pull request merged**